src/Function.cpp

```diff
@@ -58,7 +58,7 @@ class WeakenFunctionPtrs : public IRMutator2 {
 }
 
 struct FunctionContents {
-    std::string name;
+    std::string name, parent_name;
```

Comment:
Is there a better name for this? "origin_name" perhaps? Also, separate lines seems better.